package.json

```diff
@@ -73,6 +73,7 @@
     "react-hot-loader": "4.7.1",
     "react-router": "4.3.1",
     "react-router-dom": "4.3.1",
+    "react-virtualized": "9.21.0",
```

Comment:
This needs an open discussion, pagination vs infinite scroll. cc: @ayusharma

Reply:
cc: @verdaccio/maintainers @verdaccio/collaborators

Reply:
(1) Right now it gets all data at one request, i.e. without pagination, and it is 18kb response for 3k packages at my local repo (and then was rendered by List component that renders only visible items). It looks ok for me to do it this way now, but if someone will get 10 times more packages with 10 versions for any package then it can be over 1Mb, and this is not good.

(2) If paging arrives then we need "search" endpoint, I guess?

Reply:
(1) Agree. 
(2) Yes.

I'd go personally with (2), but I'd wait to hear more opinions (hopefully arrive some).

Reply:
We could stream JSON using Oboe.js or another solution.

Reply:
Do you have any example that we can take a look such implementation?

Reply:
Did not work with oboe.js in the last years.
But npm is also using it as it seems.

http://oboejs.com/examples
https://github.com/jimhigson/oboe.js/network/dependents?dependent_type=PACKAGE
https://github.com/npmtest/node-npmtest-oboe
https://github.com/npm/npmepm/blob/0a37de7aa2cd560836fe6543fd5567e361549e35/lib/dockerps.js#L19-L39

Reply:
http://oboejs.com/api

Reply:
I'd love to have pagination in verdaccio ✋